### PR TITLE
[DI-1167] Call trackLink function with instance context

### PR DESCRIPTION
**Pull request by janjaap**
This PR fixes the "TypeError: Cannot read property 'pushInstruction' of undefined" error that was thrown whenever an outbound link was clicked. The `trackLink` function didn't have any context and would thus throw that error.

### CODE REVIEW

File: packages/react/src/utils/useOutboundClickListener.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const useOutboundClickListener = ({ trackLink }: MatomoInstance): void => {
     const { target } = event
     const { nodeName } = target as HTMLElement
 
-    if (nodeName === 'A' && trackLink) {
+    if (nodeName === 'A' && matomoInstance.trackLink) {
```
Comment (jonkoops) [inline packages/react/src/utils/useOutboundClickListener.tsx:13]: Do we need to check for the presence of `matomoInstance.trackLink` here? It should always be present.

File: packages/react/src/utils/useOutboundClickListener.tsx [hunk 2]
```diff
@@ -22,13 +22,13 @@ const useOutboundClickListener = ({ trackLink }: MatomoInstance): void => {
           ),
         )
       ) {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-        trackLink({ href })
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+        matomoInstance.trackLink.call(matomoInstance, { href })
```
Comment (jonkoops) [inline packages/react/src/utils/useOutboundClickListener.tsx:26]: Why does the calling the method on the original object lose it's scope? Should the following code not work also?

```ts
matomoInstance.trackLink({ href })
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: packages/react/src/utils/useOutboundClickListener.tsx (hunk 2 above)
Comment (janjaap) [reply packages/react/src/utils/useOutboundClickListener.tsx]: My guess is that the destructuring of the `useOutboundClickListener` function parameter makes the `trackLink ` function lose its context. I've changed the way the `trackLink` function is called. Also works.

**Review by jonkoops — APPROVED** [review]
Looks good to me like this. @janjaap could you add an entry to the changelog describing what this change does/fixes?

**Review by jonkoops — APPROVED** [review]

**Pull request merged**